### PR TITLE
[chore] Enable lint for integration test files and fix issues

### PR DESCRIPTION
**Pull request by djaglowski**
A couple of error are ignored because they may be more involved fixes. I suggest these be handled as separate PRs:
- [x] #17975 
- [x] #17996
- [x] #17997
- [x] #17998

### CODE REVIEW

File: receiver/mysqlreceiver/integration_test.go [hunk 1]
```diff
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build integration
-// +build integration
-
```
Comment (dmitryax) [inline receiver/mysqlreceiver/integration_test.go:17]: should still be integration, right?

Comment (djaglowski) [reply receiver/mysqlreceiver/integration_test.go]: Yes, still iterating on this

File: receiver/riakreceiver/integration_test.go [hunk 2]
```diff
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build integration
-// +build integration
-
```
Comment (dmitryax) [inline receiver/riakreceiver/integration_test.go:17]: Shouldn't be removed?

**Review by dmitryax — APPROVED** [review]

File: receiver/riakreceiver/integration_test.go (hunk 2 above)
Comment (djaglowski) [reply receiver/riakreceiver/integration_test.go]: Good catch. Fixed.

File: receiver/riakreceiver/integration_test.go [hunk 3]
```diff
@@ -87,5 +87,5 @@ func TestRiakIntegration(t *testing.T) {
 
 	actualMetrics := consumer.AllMetrics()[0]
 
-	pmetrictest.CompareMetrics(expectedMetrics, actualMetrics, pmetrictest.IgnoreMetricValues())
+	pmetrictest.CompareMetrics(expectedMetrics, actualMetrics, pmetrictest.IgnoreMetricValues()) //nolint:errcheck
```
Comment (bogdandrutu) [inline receiver/riakreceiver/integration_test.go:90]: Why not expect no error?

**Review by bogdandrutu — APPROVED** [review]

File: receiver/riakreceiver/integration_test.go (hunk 3 above)
Comment (bogdandrutu) [reply receiver/riakreceiver/integration_test.go:90]: I think you mentioned something in the description. Fine to resolve in a separate PR.

Comment (djaglowski) [reply receiver/riakreceiver/integration_test.go:90]: You're correct that it should be that way, but see issue description. I've captured each in an issue for now.

**Pull request merged**